lib/maths/CBoostedTreeFactory.cc

```diff
@@ -73,7 +73,7 @@ std::size_t computeMaximumNumberTrees(double eta) {
     return static_cast<std::size_t>(3.0 / eta / MIN_DOWNSAMPLE_FACTOR_SCALE + 0.5);
 }
 
-std::size_t scaleMaximumNumberTrees(std::size_t maximumNumberTrees) {
+std::size_t scaleMaximumNumberTreesForMainLoop(std::size_t maximumNumberTrees) {
```

Comment:
nit: Does `ForMainLoop` adds important information to the function name?

Reply:
Well I wanted to somehow capture that this only applies in the main optimisation loop, because we're allowing the downsample factor and this is coupled with the choice of eta parameter. I think this is useful to convey somehow. Maybe I can come up with something better though.

Reply:
(I take your point about other variable naming though)

Reply:
Ok I ended up removing this method altogether. We could probably still do with some rationalisation around the maximum number of trees and how it is used in the factory class, but I'll leave that for a separate change.

Reply:
See [7a13f76](https://github.com/elastic/ml-cpp/pull/948/commits/7a13f76508cd0add593b3017bb28b1427f0d2429)